Servers/UI/OJS.Servers.Ui/ClientApp/src/components/problems/problem-resource/ProblemResource.tsx

```diff
@@ -46,24 +49,25 @@ const ProblemResource = ({ resource }: IProblemResourceProps) => {
             </Button>
         )), [ handleDownloadResourceFile, resource ]);
 
-    const resourceTypeIconClassName = useMemo(
-        () => resource.type == null
-            ? resourceTypeToIconClassName[1]
-            : resourceTypeToIconClassName[resource.type],
-        [ resource.type ],
-    );
+    const renderResourceIcon = useCallback(
+        () => {
+            const Icon = resourceTypeToIcon[resource.type] || null;
+
+            if (isNil(Icon)) {
+                return null;
+            }
 
-    const resourceIconClassName = useMemo(
-        () => concatClassNames(styles.icon, resourceTypeIconClassName),
-        [ resourceTypeIconClassName ],
+            return (<Icon size={IconSize.Medium} />);
+        },
+        [ resource.type ],
     );
 
     const getResourceLinkContent = useCallback(() => (
```

Comment:
There is no point in having a useCallback that directly returns JSX. Make renderResourceIcon() and this function into a single one.